beacon-chain/blockchain/process_block_helpers.go

```diff
@@ -247,10 +247,19 @@ func (s *Service) updateFinalized(ctx context.Context, cp *ethpb.Checkpoint) err
 	}
 
 	fRoot := bytesutil.ToBytes32(cp.Root)
+	optimistic, err := s.cfg.ForkChoiceStore.IsOptimistic(ctx, fRoot)
```

Comment:
There are some wierd edge cases here, if we have an optimistic finalized root do we still migrate our states ? Or do we actually wait

Reply:
If we finalize an optimistic block we shouldn't update LastValidatedCheckpoint.